traffic_ops/traffic_ops_golang/topology/queue_update.go

```diff
@@ -0,0 +1,100 @@
+package topology
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/apache/trafficcontrol/lib/go-tc"
+	"github.com/apache/trafficcontrol/lib/go-tc/tovalidate"
+	"github.com/apache/trafficcontrol/lib/go-util"
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api"
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/dbhelpers"
+	validation "github.com/go-ozzo/ozzo-validation"
+	"net/http"
+)
+
+func Validate(reqObj tc.TopologiesQueueUpdateRequest, topologyName tc.TopologyName, tx *sql.Tx) error {
+	errorMap := validation.Errors{}
+
+	if reqObj.Action != "queue" && reqObj.Action != "dequeue" {
+		errorMap["action"] = errors.New("action must be 'queue' or 'dequeue'")
+	}
+
+	if _, cdnExists, err := dbhelpers.GetCDNNameFromID(tx, reqObj.CDNID); err != nil {
+		errorMap["cdnId"] = fmt.Errorf("could not check whether cdn exists for id %d", reqObj.CDNID)
+	} else if !cdnExists {
+		errorMap["cdnId"] = fmt.Errorf("no cdn exists with id %d", reqObj.CDNID)
+	}
+
+	if topologyExists, err := dbhelpers.TopologyExists(tx, topologyName); err != nil {
+		errorMap["topology"] = fmt.Errorf("could not check whether topology %s exists", topologyName)
+	} else if !topologyExists {
+		errorMap["topology"] = fmt.Errorf("no topology exists by the name of %s", topologyName)
+	}
+
+	return util.JoinErrs(tovalidate.ToErrors(errorMap))
+}
+
+func QueueUpdateHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
Might be good to add a description comment here

Reply:
Description added in a3aba0f95c